cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/Schedulers.java

```diff
@@ -97,21 +100,21 @@ public static ProgramScheduleStoreDataset getScheduleStore(DatasetContext contex
     }
   }
 
-  public static ScheduleCreationSpec toScheduleCreationSpec(NamespaceId deployNamespace, Schedule schedule,
-                                                            String programName, Map<String, String> properties) {
-    Trigger trigger;
+  public static ScheduleCreationBuilder toScheduleCreationBuilder(NamespaceId deployNamespace, Schedule schedule,
+                                                                  String programName, Map<String, String> properties) {
+    TriggerBuilder triggerBuilder;
     if (schedule instanceof TimeSchedule) {
-      trigger = new TimeTrigger(((TimeSchedule) schedule).getCronEntry());
+      triggerBuilder = new TimeTriggerBuilder(((TimeSchedule) schedule).getCronEntry());
     } else {
       StreamSizeSchedule streamSizeSchedule = ((StreamSizeSchedule) schedule);
-      trigger = new StreamSizeTrigger(deployNamespace.stream(streamSizeSchedule.getStreamName()),
-                                      streamSizeSchedule.getDataTriggerMB());
+      triggerBuilder = new StreamSizeTriggerBuilder(deployNamespace.stream(streamSizeSchedule.getStreamName()),
+                                                    streamSizeSchedule.getDataTriggerMB());
     }
     Integer maxConcurrentRuns = schedule.getRunConstraints().getMaxConcurrentRuns();
     List<Constraint> constraints = maxConcurrentRuns == null ? ImmutableList.<Constraint>of() :
       ImmutableList.<Constraint>of(new ConcurrencyConstraint(maxConcurrentRuns));
-    return new ScheduleCreationSpec(schedule.getName(), schedule.getDescription(), programName,
```

Comment:
@maochf: Is the first constructor in [`ScheduleCreationSpec`](https://github.com/caskdata/cdap/blob/develop/cdap-api/src/main/java/co/cask/cdap/internal/schedule/ScheduleCreationSpec.java#L41) necessary anymore? This seems to be only place it was being used